### DISCUSSION TITLE
Help needed to simulate a Vernier Synchronous Machine

### DISCUSSION THREAD

**Discussion by Loodt** [Eomys/pyleecan#663] [Q&A]
I need help on the next steps. I am trying to simulate a Vernier Synchronous machine. However, I am struggling to follow the tutorials on different machines that are not mentioned in the tutorials. 

So I started modelling the machine using the MachineUD class as in the first tutorial. However, the next tutorial uses the IPMSM machine and the MachineUD doesn't have a rotor object.

Can someone please point me in the right direction on where to start? How do I run FEMM analyses on my machine? As far as I understand I still need to assign the magnet material properties and then the electrical properties of the machine. Before I can simulate. I just keep on running into errors and I think I started with the wrong Machine class.

Please see my code below:
[pmvm_v10.zip](https://github.com/Eomys/pyleecan/files/12996831/pmvm_v10.zip)

**Comment by BonneelP** [comment]
Hello,

Thank you for your interest in pyleecan.
Can you share drawings of the machine you plan to simulate so we can discuss on the best strategy to design it ?
If I understand your needs, maybe the latest feature of keys (cf #658) can help you to add ferromagnetic parts in the airgap ?

Best regards,
Pierre

**Reply by Loodt** [reply]
Hi Pierre,

Thank you for getting back to me so quickly. Please take a look at the structure the typical vernier structure below in the image. This image was taken from a study: "Design and Performance Comparison of Vernier and Conventional PM Synchronous Wind Generators".

The vernier machine has a magnetic gearing effect, which gives it a very high torque density. I was planning first to simulate structure (a) on the image and would like to be able to do (b) as well.

I think adding ferromagnetic parts in the airgap could work.
![PMVM tructures](https://github.com/Eomys/pyleecan/assets/31105647/a4003855-22df-4511-93da-6347f7cc2453)

**Reply by BonneelP** [reply]
Hello,

Thank you for the image (I'm not used to this kind of machine). If I understand correctly, both machine should be possible in the GUI by using an outer rotor SPMSM. 
(a) stator looks normal and for the rotor:
- if red and blue are north/south magnets it's straightforward 
- if one of them is not a magnet then you will need to add notches with key which is possible in the GUI (on the master branch, not in the release yet)
- if none of them are magnets then it may require some new development (then you can define the machine with magnets instead and edit the femm file to show us what should be the properties of the surface).
For the stator, if you need to use Slot 16 there is a bug that we have corrected internally but not shared yet (we plan to include that in a release early November)

(b) same for the rotor, for the stator, you can define a slot with Zs=6 for the winding + 3 sets of notches Zs=6 for the teeth (can be done in the GUI in the release).

Do you think that you can share your machine and results once you manage to run them ? I would be a great validation case for pyleecan in particular if compared with a publication :)

Best regards,
Pierre

**Reply by Loodt** [reply]
Thank you, Pierre. Yes, I will share the results if successful.

**Reply by nkbw33** [reply]
Hi Loodt, were you successful in modeling the vernier machine?
